UofT-Frosh-Orientation/orientation-website
# 648 update QR code logic

## **Contribution:**


### **Issue:**

Closes #648 

### **Accomplishments:**

- Made QR codes work with  user ID only
- changesd frontend and backend code to works with the user ID instead of email
- added some better error handeling 
### **Visuals:** 

Any related visuals.

## **Details:**

### **Expected Behavior:**

- QR codes in profile page and PDF contain the string for the users ID
- Scanning he QR code  will extract the ID
- The backend `qr/scan` route accepts the new QR code

### **Testing Steps:**

Steps required to get expected behavior.

- Make admin account
- Make frosh account
- Pull up QR code on phone (pdf or profile page)
- Start scanning on admin account
- Scanning should work as before

### **Complications:**

- Found out that the QR code in the PDF didnt have the proper data since the `froshObject` passed into it didnt have an ID attribute.
